src/stock-orderable-group/orderable-group.service.js

```diff
@@ -0,0 +1,292 @@
+ * version 3 of the License, or (at your option) any later version.
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+(function() {
+
+    'use strict';
+
+    /**
+     * @ngdoc service
+     * @name stock-orderable-group.orderableGroupService
+     *
+     * @description
+     * Responsible for managing orderable groups.
+     */
+    angular
+        .module('stock-orderable-group')
+        .service('orderableGroupService', service);
+
+    service.$inject = ['messageService', 'StockCardSummaryRepositoryImpl',
+        'FullStockCardSummaryRepositoryImpl', 'StockCardSummaryRepository', 'dateUtils'];
+
+    function service(messageService, StockCardSummaryRepositoryImpl,
+                     FullStockCardSummaryRepositoryImpl, StockCardSummaryRepository, dateUtils) {
+
+        var noLotDefined = {
+            lotCode: messageService.get('orderableGroupService.noLotDefined')
+        };
+
+        this.findAvailableProductsAndCreateOrderableGroups = findAvailableProductsAndCreateOrderableGroups;
+        this.lotsOf = lotsOf;
+        this.determineLotMessage = determineLotMessage;
+        this.groupByOrderableId = groupByOrderableId;
+        this.findByLotInOrderableGroup = findByLotInOrderableGroup;
+        this.areOrderablesUseVvm = areOrderablesUseVvm;
+        this.getKitOnlyOrderablegroup = getKitOnlyOrderablegroup;
+        this.addItemWithNewLot = addItemWithNewLot;
+
+        /**
+         * @ngdoc method
+         * @methodOf stock-orderable-group.orderableGroupService
+         * @name lotsOf
+         *
+         * @description
+         * Extract lots from orderable group. Adds no lot defined as an option when some group
+         * has no lot
+         *
+         * @param {Object} orderableGroup   orderable group
+         * @return {Array}                  array with lots
+         */
+        function lotsOf(orderableGroup, addMissingLotAllowed) {
+            var addMissingLot = {
+                    lotCode: messageService.get('orderableGroupService.addMissingLot')
+                },
+                lots;
+
+            if (orderableGroup && orderableGroup.length > 0 && orderableGroup[0].$allLotsAdded) {
+                lots = [];
+            } else {
+                lots = _.chain(orderableGroup).pluck('lot')
+                    .compact()
+                    .value();
+
+                lots.forEach(function(lot) {
+                    lot.expirationDate = dateUtils.toDate(lot.expirationDate);
+                });
+
+                // ANGOLASUP-516: Removed the 'No Lot Defined' option
```

Comment:
please add another comment which indicates where the customization ends, ex ANGOLASUP-516: ends here

Reply:
It is related to the 'removed' code, not 'added' one (as we discussed on slack)